editor/src/clj/editor/model_scene.clj

```diff
@@ -300,23 +302,166 @@
       default-material-ids
       ret)))
 
-(defn- make-renderable-mesh [mesh mesh-material-index->material-name]
+(defn- vec->vector4d [v]
+  (Vector4d. (nth v 0) (nth v 1) (nth v 2) (nth v 3)))
+
+(defn- make-renderable-material-data [mesh-material-data]
+  (when mesh-material-data
+    (let [pbr-metallic-roughness (:pbr-metallic-roughness mesh-material-data)
+          pbr-specular-glossiness (:pbr-specular-glossiness mesh-material-data)
+          pbr-clear-coat (:clearcoat mesh-material-data)
+          pbr-transmission (:transmission mesh-material-data)
+          pbr-ior (:ior mesh-material-data)
+          pbr-specular (:specular mesh-material-data)
+          pbr-volume (:volume mesh-material-data)
+          pbr-sheen (:sheen mesh-material-data)
+          pbr-emissive-strength (:emissive-strength mesh-material-data)
+          pbr-iridescence (:iridescence mesh-material-data)
+          pbr-texture-index->value (fn ^double [^long ix]
+                                     (if (>= ix 0) 1.0 0.0))]
```

Comment:
Not sure I understand this. Should it be the index instead of `1.0`?

Reply:
Hm no, because we can't really do anything with indices in the shader. We can't bind textures as arrays, so the value simply means if the value should be gathered from a specific sampler.